File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/minimum-number-of-swaps-to-make-the-string-balanced.java
class Solution {
    public int minSwaps(String s) {
        int balance = 0;
        int maxUnbalanced = 0;

        for (char c : s.toCharArray()) {
            if (c == '[') {
                balance++;
            } else { 
                balance--;
                if (balance < 0) {
                    maxUnbalanced = Math.max(maxUnbalanced, -balance);
                }
            }
        }

        return (maxUnbalanced + 1) / 2;
    }
}

File: 2095-minimum-number-of-swaps-to-make-the-string-balanced/minimum-number-of-swaps-to-make-the-string-balanced.py
class Solution:
    def minSwaps(self, s: str) -> int:
        stack = []
        for char in s:
            if(char == '['):
                stack.append('[')
            else:
                if(len(stack)==0 or stack[-1]!='[' ):
                    stack.append(']')
                else:
                    stack.pop()
        ans=0
        a=stack
        print(a)
        if(len(a)%2!=0):
            ans=-1
        elif(len(a)==0):
            return 0
        else:
            i=0
            j=len(a)-1
            ans=0
            while(i<j):
                if(a[i]=="]" and a[j]=="["):
                    ans+=1
                i+=1
                j-=1
        return (ans+1)//2
        